### PR TITLE
Clarify optimade versioning

### PR DESCRIPTION
**Pull request by rartino**
It was requested in the release planning for v1.2 #429 that we clarify how semantic versioning applies to the versioning of the OPTIMADE standard. I've made my best attempt to be as clear as possible on what I believe is/should be our policy.

### CODE REVIEW

File: optimade.rst [hunk 1]
```diff
@@ -226,6 +226,28 @@ The definition of a property of an entry type specifies a type. The value of tha
...
+
+Working copies distributed as part of the development of the standard are marked with the version number for the release they are based on with an additional "~develop" suffix.
+These "versions" do not refer to a single specific instance of the text (i.e., the same "~develop" version string is retained until a release), nor is it clear to what degree they contain backwards incompatible API changes.
+Hence, the suffix is intentionally designed to make these version strings not to conform with semantic versioning to prevent incorrect comparisons to released versions using the scheme prescribed by semantic versioning.
+Version strings with a "~develop" suffix MAY be used by implementations during testing.
+However, a client that encounter them unexpectedly SHOULD NOT make any assumptions about the level of API compatibility.
+
+In conclusion, the versioning policy of this standard is designed to allow clients using the OPTIMADE API according to a specific version of the standard to assume compatibility with servers implementing any future version of the standard sharing the same MAJOR version number.
```
Comment (rartino) [inline optimade.rst:249]:
```suggestion
In conclusion, the versioning policy of this standard is designed to allow clients using the OPTIMADE API according to a specific version of the standard to assume compatibility with servers implementing any future (non-development) version of the standard sharing the same MAJOR version number.
```

**Review by ml-evs — DISMISSED** [review]
Thanks @rartino, this reads well to me.

File: optimade.rst [hunk 2]
```diff
@@ -226,6 +226,28 @@ The definition of a property of an entry type specifies a type. The value of tha
 General API Requirements and Conventions
 ========================================
 
+Versioning of this standard
+---------------------------
+This standard describes a communication protocol that, when implemented by a server, provides clients with an API for data access.
+
+Released versions of the standard are versioned using `semantic versioning v2 <https://semver.org/spec/v2.0.0.html>`__ in reference to changes in *that API* (i.e., not in the server-side implementation of the protocol).
+
+To clarify: semantic versioning mandates version numbers on the form MAJOR.MINOR.PATCH, where a "backwards incompatible API change" requires incrementing the MAJOR version number.
```
Comment (merkys) [inline optimade.rst:235]:
```suggestion
To clarify: semantic versioning mandates version numbers of the form MAJOR.MINOR.PATCH, where a "backwards incompatible API change" requires incrementing the MAJOR version number.
```

**Review by merkys — APPROVED** [review]

**Comment by rartino** [comment]
Pinging @ml-evs on a quick revist on this as well.

**Review by ml-evs — APPROVED** [review]

**Pull request merged**